source/extensions/filters/common/lua/lua.cc

```diff
@@ -22,7 +22,7 @@ void Coroutine::start(int function_ref, int num_args, const std::function<void()
 
   state_ = State::Yielded;
   lua_rawgeti(coroutine_state_.get(), LUA_REGISTRYINDEX, function_ref);
-  ASSERT(lua_isfunction(coroutine_state_.get(), -1));
+  ASSERT(lua_isfunction(coroutine_state_.get(), -1) == 1);
```

Comment:
Can you comment why this needs to be exactly 1? :)

Reply:
Hmm it may not be. Taking this out